src/ReflectionFile.php

```diff
@@ -50,6 +50,9 @@ class ReflectionFile
      */
     public function __construct($fileName, $topLevelNodes = null)
     {
+        if (!is_string($fileName)) {
```

Comment:
The only reason I made this addition was that it was flagged by Scrutinizer:

> Documentation Bug introduced 30 days ago by @lisachenko
> It seems like `$fileName` can also be of type `array` or `Boolean`. However, the property `$fileName` is declared as type `string`. Maybe add an additional type check?

I think this based on type variations in the calling functions, despite this function only ever being called with string. I expect the other change that might fix this is isolating the parameter into a distinct, string-only variable that is used when calling this method. I think this type check is the easier solution.